compiler/rustc_const_eval/src/check_consts/check.rs

```diff
@@ -360,6 +361,73 @@ impl<'mir, 'tcx> Checker<'mir, 'tcx> {
         // end of evaluation.
         !is_transient
     }
+
+    fn revalidate_conditional_constness(
+        &mut self,
+        callee: DefId,
+        callee_args: ty::GenericArgsRef<'tcx>,
+        call_source: CallSource,
+        call_span: Span,
+    ) {
+        let tcx = self.tcx;
+        if !tcx.is_conditionally_const(callee) {
+            return;
+        }
+
+        let const_conditions = tcx.const_conditions(callee).instantiate(tcx, callee_args);
+        // If there are any const conditions on this fn and `const_trait_impl`
+        // is not enabled, simply bail. We shouldn't be able to call conditionally
+        // const functions on stable.
+        if !const_conditions.is_empty() && !tcx.features().const_trait_impl() {
```

Comment:
You should never call `tcx.features()` in this code, as that won't properly check for `rustc_allow_const_fn_unstable`.

Reply:
Well, the user still needs to enable the feature gate *globally* so that we can properly enforce the feature's validity earlier in the compiler.